File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/Java/Go/s824230885_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

type InputIterator struct {
	inputLine []string
	index     int
	max       int
}

func NewInputIterator() *InputIterator {
	reader := bufio.NewReader(os.Stdin)
	inputLine := make([]string, 0)
	for {
		line, err := reader.ReadString('\n')
		if err != nil {
			break
		}
		inputLine = append(inputLine, strings.TrimSpace(line))
	}
	return &InputIterator{
		inputLine: inputLine,
		index:     0,
		max:       len(inputLine),
	}
}

func (ii *InputIterator) HasNext() bool {
	return ii.index < ii.max
}

func (ii *InputIterator) Next() string {
	if ii.HasNext() {
		result := ii.inputLine[ii.index]
		ii.index++
		return result
	}
	panic("There is no more input")
}

func nextInt(ii *InputIterator) int {
	return strconv.Atoi(ii.Next())
}

func nextIntArray(ii *InputIterator) []int {
	input := strings.Fields(ii.Next())
	result := make([]int, len(input))
	for i, s := range input {
		result[i], _ = strconv.Atoi(s)
	}
	return result
}

func solve() {
	ii := NewInputIterator()
	N := nextInt(ii)
	list := nextIntArray(ii)
	max := 0
	output := 0
	for i := 0; i < N; i++ {
		if max <= list[i] {
			max = list[i]
			output++
		}
	}
	fmt.Println(output)
}

func main() {
	solve()
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/Java/Python/s824230885_commented.py
import sys
from typing import List

class InputIterator:
    def __init__(self):
        self.input_line = []
        for line in sys.stdin:
            self.input_line.append(line.strip())
        self.index = 0
        self.max = len(self.input_line)

    def hasNext(self) -> bool:
        return self.index < self.max

    def next(self) -> str:
        if self.hasNext():
            return self.input_line[self.index]
        else:
            raise IndexError("There is no more input")

    def nextInt(self) -> int:
        return int(self.next())

    def nextLong(self) -> int:
        return int(self.next())

    def nextDouble(self) -> float:
        return float(self.next())

    def nextStrArray(self) -> List[str]:
        return self.next().split()

    def nextCharArray(self) -> List[str]:
        return list(self.next())

    def nextIntArray(self) -> List[int]:
        return [int(x) for x in self.nextStrArray()]

    def nextLongArray(self) -> List[int]:
        return [int(x) for x in self.nextStrArray()]

def myout(t):
    print(t)

def myerr(t):
    print(f"debug: {t}", file=sys.stderr)

def solve():
    ii = InputIterator()
    N = ii.nextInt()
    list_ = ii.nextIntArray()
    max_ = 0
    output = 0
    for i in range(N):
        if max_ <= list_[i]:
            max_ = list_[i]
            output += 1
    myout(output)

solve()  # End of Code
